web-server/src/controllers/auth/register.ts

```diff
@@ -1,60 +1,56 @@
 import { Request, Response, NextFunction } from 'express';
-
 import { AppDataSource } from 'orm/data-source';
 import { User } from 'orm/entities/users/User';
-import { CustomError } from 'utils/response/custom-error/CustomError';
-import { ErrorArray } from 'utils/response/custom-error/types';
+import { UserError } from 'utils/Errors';
 
 export const register = async (req: Request, res: Response, next: NextFunction) => {
-  const errors = new ErrorArray();
-
-  const { email, username, password } = req.body;
+  const { email, username, password, passwordConfirm } = req.body;
 
   const userRepository = AppDataSource.getRepository(User);
+  
+  const err: UserError = {};
+
   try {
+
     let user = await userRepository.findOne({ where: { email } });
 
     if (user) {
-      errors.put('email', `User with email ${email} already exists`);
+      err.email = `User with email "${email}" already exists`
     }
 
     user = await userRepository.findOne({ where: { username } });
 
     if (user) {
-      errors.put('username', `User with username ${username} already exists`);
+      err.username = `User with username "${username}" already exists`;
     }
 
+    user = new User();
+    user.email = email;
+
     try {
-      user = new User();
-      user.email = email;
-
-      try {
-        user.setUsername(username);
-      } catch (err: unknown) {
-        if (err instanceof CustomError) {
-          errors.extend(err.JSON.errors);
-        } else {
-          const customError = new CustomError(400, 'Raw', 'Error', err, errors);
-          return next(customError);
-        }
+      user.setUsername(username);
+    } catch (e) {
+      if ('username' in e) {
+        err.username = e.username;
+      } else {
+        err.raw = e;
       }
+    }
 
-      user.hashedPassword = password;
-      user.hashPassword();
+    user.hashedPassword = password;
+    user.hashPassword();
```

Comment:
This is the one opposite case.

Instead we should do this:
```
class User {
  ...
  setPassword(password:string) {
    this.hashedpassword = hash(password);
  }
  ...
}

user.setPassword(password);
```

The difference is that if you mess up and call `user.hashPassword()` twice, you end up breaking your user's login due to how hashing works.

Also it's unelegant because this is clearly a two-step process to assign a value unlike the other case where it's really just validation.